Page Block Tabs (#397)
cortezaproject/corteza-docs: src/modules/integrator-guide/pages/compose-configuration/page-blocks.adoc

Defines the display element from the selected report the page block will use when showing data.

|===

[#page-block-tabs]
== Tabs

The *tabs* page block allows the grouping of different page blocks together.
Sometimes a page might contain different blocks and simply laying them out side by side might not be the best way to display them.

> Tabs allow you to have the information you need without cluttering the page, making it easier to navigate a page and being more productive.

Reviewer: I feel like the first and the third sentence say the same thing; can we merge them together?

|===

[#page-block-tabs]
== Tabs

The *tabs* page block allows the grouping of different page blocks together, decluttering, and making it easier to navigate a page.
Sometimes a page might contain different blocks and simply laying them out side by side might not be the best way to display them.

> === Configuring Tabs

Reviewer: new line before the title

> You can add a *tabs* page block to a page by clicking on the *Add* button and selecting *Tabs* from the list of available page blocks.

Reviewer: ```suggestion
You can add a *tabs* page block to a page by clicking on the btn:[Add] button and selecting *Tabs* from the list of available page blocks.
```

In the *tabs* tab of its configurator, you can add your desired page blocks to it or refine the appearance of the tabs page block itself.

> To tab a page block, click on the *+ Add* button and select the desired page block from the list of available page blocks, if you added blocks to the page already.

Reviewer: ```suggestion
To tab a page block, click on the btn:[+ Add] button and select the desired page block from the list of available page blocks, if you added blocks to the page already.
```

.Tabs page block configuration:
[annotation,role="data-zoomable"]
----
{
  "image": {
    "rel": "compose-configuration/config-tabs2.png",
    "alias": "config-tabs-add-button-block-selector",
    "w": 1920,
    "h": 1080
  },
  "view": {},
  "focus": {
    "x": 410,
    "y": 50,
    "w": 1100,
    "h": 865
  },
  "annotations": [
    {
      "x": 410,
      "y": 611,
      "w": 75,
      "h": 30
    },
    {
      "x": 938,
      "y": 415,
      "w": 490,
      "h": 40
    }
  ]
}
----

Before tabbing a page block, you will find an input box, this controls the title of the tab. 
If not set, the title of the page block is used, and if that is unset, a generic name is used which is based on the *kind* of the page block.

.Tabs page block configuration:
[annotation,role="data-zoomable"]
----
{
  "image": {

>     "rel": "compose-configuration/config-tabs2.png",

Reviewer: can we pick a more descriptive name then `config-tabs2`?

    "alias": "config-tabs-title-input-box",
    "w": 1920,
    "h": 1080
  },
  "view": {},
  "focus": {
    "x": 410,
    "y": 50,
    "w": 1100,
    "h": 865
  },
  "annotations": [
    {
      "x": 438,
      "y": 415,
      "w": 490,
      "h": 40
    }
  ]
}
----

> Although a tabbed block is removed from the main page, you can still access its configuration via the *edit* button corresponding to it in the tabs page block's configurator.

Reviewer: ```suggestion
Although a tabbed block is removed from the main page, you can still access its configuration via the btn:[edit] button corresponding to it in the tabs page block's configurator.
```

.Tabs page block configuration:
[annotation,role="data-zoomable"]
----
{
  "image": {
    "rel": "compose-configuration/config-tabs2.png",
    "alias": "config-tabs-delete-tab-button",
    "w": 1920,
    "h": 1080
  },
  "view": {},
  "focus": {
    "x": 410,
    "y": 50,
    "w": 1100,
    "h": 865
  },
  "annotations": [
    {
      "x": 1462,
      "y": 519,
      "w": 15,
      "h": 15
    }
  ]
}
----

=== Tabs Caveats

Tabbing a block removes it from the main page.
You can override this behavior by going to the general tab of that page block and disabling the 'hidden' option for the block.
This removes it from wherever it is tabbed and returns it to being displayed on the main page.

.Generic page block configuration:
[annotation,role="data-zoomable"]
----
{
  "image": {
    "rel": "compose-configuration/pageblock-hidden.png",
    "alias": "config-pageblock-hidden-option",
    "w": 1920,
    "h": 1080
  },
  "view": {},
  "focus": {
    "x": 410,
    "y": 50,
    "w": 1100,
    "h": 865
  },
  "annotations": [
    {
      "x": 410,
      "y": 476,
      "w": 100,
      "h": 18
    }
  ]
}
----

[NOTE]
====

> When you delete a tab block that holds other page blocks, how many times those blocks occur in other *tabs* block will be determined and they will be added to the main page or not based on the result of that computation.

Reviewer: Can we reword this; it's a bit confusing to read

|===

[#page-block-tabs]
== Tabs

The *tabs* page block allows the grouping of different page blocks together, decluttering, and making it easier to navigate a page.

> === Configuring Tabs

Reviewer: This doesn't belong here.
Page block reference provides a glance over the available page blocks.
If we want to go into details how to use something, how to add this thing, etc. move it somewhere else in the low-code config sections.

Since this mostly talks about how to add the page block, I'd just remove it since this is discussed elsewhere and not that complicated to re-explain it here.

> === Configuring Tabs

Reviewer: See the note on this not belonging here. This is just a reference

To tab a page block, select the desired page block from the list of available page blocks.

.Tabs page block configuration:
[annotation,role="data-zoomable"]
----
{
  "image": {

>     "rel": "compose-configuration/config-tabs-add-button.png",

Reviewer: The blur is still off; please tweak it.
Also, I'm not sure the annotations you've added match up... They don't make sense in my opinion so I'd remove it.

  "image": {

>     "rel": "compose-configuration/config-tabs-add-button.png",

Reviewer: Your annotations are a bit off; please be more accurate.
It doesn't need to be pixel-perfect but do try to be as accurate as possible (it doesn't take a lot of effort with those _tools I added_).
Also, notice the blur; it's a bit off at the bottom.

![image](https://github.com/cortezaproject/corteza-docs/assets/12545711/a3788ae7-185b-45ea-a592-28d2b0369235)

    "h": 1080
  },
  "view": {},
  "focus": {
    "x": 410,
    "y": 50,
    "w": 1100,
    "h": 865
  },
  "annotations": [
    {
      "x": 412,
      "y": 612,
      "w": 73,
      "h": 29
    },
    {
      "x": 938,
      "y": 417,
      "w": 488,
      "h": 36
    }
  ]
}
----

.Configurator Options:
[cols="2s,5a"]
|===
| [#page-block-tabs-add]#<<page-block-tabs-add,Add Tab>>#
|
Allows you to create a new tab.

[NOTE]
====

> Tabbing a block removes it from the main page.

Reviewer: ```suggestion
Adding a page block to the tabbed page block removes it from the main page.
```

> [NOTE]

Reviewer: Empty line above `[NOTE]`

====

| [#page-block-tabs-title]#<<page-block-tabs-title,Title>>#
|
Determines the title of the tab.

> If not set, the title of the page block is used, and if that is unset, a generic name is used which is based on the kind of the page block.

Reviewer: ```suggestion
If not set, the title of the page block is used, and if that is not set, a generic name based on the kind of the page block is used.
```

> Determines the title of the tab. If not set, the title of the page block is used, and if that is unset, a generic name is used which is based on the kind of the page block.

Reviewer: One sentence per line

| [#page-block-tabs-edit]#<<page-block-tabs-edit,Edit>>#
|

> Allows you to access the configurator of a tabbed block.

Reviewer: Perhaps this would be a tad bit better?

```suggestion
Allows you to access the page block configurator of a tabbed block.
```

| [#page-block-tabs-plus]#<<page-block-tabs-plus,Add Block>>#
|

> Allows you to add a new block that is immediately tabbed. Access this configuration by clicking on the btn:[+ add] button.

Reviewer: One.
Sentence.
Per.
Line.

> Allows you to add a new block that is immediately tabbed. Access this configuration by clicking on the btn:[+] button.

Reviewer: ```suggestion
Allows you to add a new block that is immediately tabbed. Access this configuration by clicking on the btn:[+ add] button.
```

> | [#page-block-tabs-delete]#<<page-block-tabs-delete,Delete>>#

Reviewer: The configurator options down here are page block specific, this is a generic thing (deleting page blocks) so it doesn't belong here

Author: Added it there because the [Note] is important

|
Allows you to delete a tabbed block.
[NOTE]
====

> When you delete a tabs block, the number of times its blocks occur in other tabs blocks will be computed, and this will determine if the blocks it held will be added to the main page or not based on the result of that computation.

Reviewer: Omit low-level bits from here; simply point out what happens if a condition is met.
All these counters and checking something and doing something is IMO boilerplate.
Just give me the tea man!! What happens if something happens??

* Dropdown: This type of navigation item has a dropdown option. When clicked, it will display a dropdown with the corresponding dropdown items.

|===

> 

Reviewer: just one empty line here please

[#page-block-tabs]
== Tabs

> The *tabs* page block allows the grouping of different page blocks together, decluttering, and making it easier to navigate a page.

Reviewer: ```suggestion
The *tabs* page block allows the grouping of different page blocks to declutter and make it easier to navigate a page.
```

I think this sounds a bunch cleaner
